### PR TITLE
apkpatch 1.0.3 (new formula)

### PR DESCRIPTION
**Pull request by geek5nan**
add  AndFix-apkpatch-1.0.3,a famous third part library's command line for Android Developers

### CODE REVIEW

**Comment by bfontaine** [comment]
Can `apkpatch` be used outside of AndFix?

**Comment by geek5nan** [comment]
Yes,`apapatch` can be used outside of AndFix.
Below is a `apkpatch` usage from [AndFix HomePage](https://github.com/alibaba/AndFix),please have a look.

> How to use?
> Prepare two android packages, one is the online package, the other one is the package after you fix bugs by coding.
 >Generate .apatch file by providing the two package,
```
usage: apkpatch -f <new> -t <old> -o <output> -k <keystore> -p <***> -a <alias> -e <***>
 -a,--alias <alias>     keystore entry alias.
 -e,--epassword <***>   keystore entry password.
 -f,--from <loc>        new Apk file path.
 -k,--keystore <loc>    keystore path.
 -n,--name <name>       patch name.
 -o,--out <dir>         output dir.
 -p,--kpassword <***>   keystore password.
 -t,--to <loc>          old Apk file path.
```

File: Library/Formula/apkpatch.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+class Apkpatch < Formula
+  desc "AndFix provides a patch-making tool called apkpatch."
```
Comment (bfontaine) [inline Library/Formula/apkpatch.rb]: This should be a description of `apkpatch`. Something like `desc "Patch-making tool for Android apps"` might work.

File: Library/Formula/apkpatch.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  homepage "https://github.com/alibaba/AndFix"
+  url "https://raw.githubusercontent.com/alibaba/AndFix/master/tools/apkpatch-1.0.3.zip"
+  sha256 "0f94cd2a675af89ff3990f4b6490dd70877ad4eb109f0cdbefece1f9a73e4512"
+
+  def install
+    # Remove Windows scripts
+    rm_rf Dir["*.bat"]
+
+    # Install files
```
Comment (bfontaine) [inline Library/Formula/apkpatch.rb]: You can remove this comment.

File: Library/Formula/apkpatch.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    prefix.install_metafiles
```
Comment (bfontaine) [inline Library/Formula/apkpatch.rb]: You don’t need this; Homebrew does it for you.

File: Library/Formula/apkpatch.rb [hunk 4]
```diff
@@ -0,0 +1,23 @@
+class Apkpatch < Formula
+  desc "a tool for build/merge Android Patch file(.apatch)"
```
Comment (bfontaine) [inline Library/Formula/apkpatch.rb]: Please remove the indefinite article: `desc "Tool for build/merge Android Patch files (.apatch)"`

Comment (geek5nan) [inline Library/Formula/apkpatch.rb]: Thank you for guiding me to complete this formula : )

File: Library/Formula/apkpatch.rb [hunk 5]
```diff
@@ -0,0 +1,21 @@
+class Apkpatch < Formula
+  desc "Tool for build/merge Android Patch file(.apatch)"
+  homepage "https://github.com/alibaba/AndFix"
+  url "https://raw.githubusercontent.com/alibaba/AndFix/master/tools/apkpatch-1.0.3.zip"
+  sha256 "0f94cd2a675af89ff3990f4b6490dd70877ad4eb109f0cdbefece1f9a73e4512"
+
+  def install
+    rm_rf Dir["*.bat"]
+
+    chmod 0755, Dir["*"]
```
Comment (bfontaine) [inline Library/Formula/apkpatch.rb]: Why is this necessary?

File: Library/Formula/apkpatch.rb [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+    libexec.install Dir["*"]
+
+    Dir.glob("#{libexec}/*.sh") do |script|
+      bin.install_symlink script => File.basename(script, ".sh")
+    end
+  end
+
+  test do
+    system "#{bin}/apkpatch", "help"
```
Comment (bfontaine) [inline Library/Formula/apkpatch.rb]: Can’t we write a test that does something more substantial than e.g. `help`/`version`? Could we e.g. create a dummy `.apatch` file and apply it on a sample Android app?

File: Library/Formula/apkpatch.rb (hunk 4 above)
Comment (bfontaine) [inline Library/Formula/apkpatch.rb]: Thank *you* for contributing to Homebrew :wink:

File: Library/Formula/apkpatch.rb (hunk 6 above)
Comment (geek5nan) [inline Library/Formula/apkpatch.rb]: The `apkpatch` just a tool using for build or merge `.apatch` files ,apply these `.aptach`'s should to integrate `AndFIx` into our android project and use it's API.

File: Library/Formula/apkpatch.rb (hunk 5 above)
Comment (geek5nan) [inline Library/Formula/apkpatch.rb]: The line `chmod 0755, Dir["*"]` is not necessary,I will remove it.

File: Library/Formula/apkpatch.rb [hunk 7]
```diff
@@ -0,0 +1,19 @@
+class Apkpatch < Formula
+  desc "Tool for build/merge Android Patch file(.apatch)"
+  homepage "https://github.com/alibaba/AndFix"
+  url "https://raw.githubusercontent.com/alibaba/AndFix/master/tools/apkpatch-1.0.3.zip"
+  sha256 "0f94cd2a675af89ff3990f4b6490dd70877ad4eb109f0cdbefece1f9a73e4512"
+
```
Comment (bfontaine) [inline Library/Formula/apkpatch.rb]: Just checked the source and it appears `akpatch` needs Java to work. Please add `depends_on :java` here. It should look like this:

```rb
  # ...
  sha256 "0f94cd2a675af89ff3990f4b6490dd70877ad4eb109f0cdbefece1f9a73e4512"

  depends_on :java

  def install
    # ...
```

Comment (geek5nan) [inline Library/Formula/apkpatch.rb]: get it.

File: Library/Formula/apkpatch.rb [hunk 8]
```diff
@@ -0,0 +1,21 @@
+class Apkpatch < Formula
+  desc "Tool for build/merge Android Patch file(.apatch)"
```
Comment (mikemcquaid) [inline Library/Formula/apkpatch.rb]: space after `file(`

File: Library/Formula/apkpatch.rb (hunk 6 above)
Comment (mikemcquaid) [inline Library/Formula/apkpatch.rb]: @geek5nan Would be good to actually apply an `.apatch` file if possible.

File: Library/Formula/apkpatch.rb (hunk 8 above)
Comment (geek5nan) [inline Library/Formula/apkpatch.rb]: OK.

File: Library/Formula/apkpatch.rb (hunk 6 above)
Comment (mikemcquaid) [inline Library/Formula/apkpatch.rb]: @geek5nan Thoughts?

**Comment by bfontaine** [comment]
Ping re Mike’s comment above?

**Comment by bfontaine** [comment]
Closing due to no response. Feel free to open another PR addressing the comments above. Thanks!
